src/app/shared/state/user/user.reducer.ts

```diff
@@ -37,7 +38,14 @@ export default function reducer(state = initialState, action: ActionWithPayload<
     }
 
     case UserActions.CLEAR_USER: {
-      return initialState;
+      return cloneDeep(initialState);
```

Comment:
Do we need cloneDeep here?

Reply:
yes actually we need to perform a clone here otherwise we pass the reference of initialState and will make errors and bugs

Reply:
Again if we get errors somewhere we should find the problem with our code, cloning initialState here is meaningless because we shouldn't manipulate user state without cloning the state before manipulation. So we clone it once here and we clone it again when we are changing the state. As you can see in other store reducers we don't clone initial state. We should fix the problem.